src/NBitcoin/Consensus.cs

```diff
@@ -3,133 +3,107 @@
-
-        /// <summary>
-        /// The reward that goes to the miner when a block is mined using proof-of-stake.
-        /// </summary>
-        public Money ProofOfStakeReward { get; set; }
-
-        /// <summary>
-        /// Maximal length of reorganization that the node is willing to accept, or 0 to disable long reorganization protection.
-        /// </summary>
-        public uint MaxReorgLength { get; set; }
-
-        /// <summary>
-        /// The maximum amount of coins in any transaction.
-        /// </summary>
-        public long MaxMoney { get; set; }
+        /// <inheritdoc />
+        public Money PremineReward { get; }
 
-        public ConsensusOptions Options { get; set; }
+        /// <inheritdoc />
+        public long PremineHeight { get; }
 
-        public class BuriedDeploymentsArray
-        {
-            private readonly int[] heights;
-
-            public BuriedDeploymentsArray()
-            {
-                this.heights = new int[Enum.GetValues(typeof(BuriedDeployments)).Length];
-            }
-
-            public int this[BuriedDeployments index]
-            {
-                get => this.heights[(int)index];
-                set => this.heights[(int)index] = value;
-            }
-        }
+        /// <inheritdoc />
+        public Money ProofOfWorkReward { get; }
 
-        public class BIP9DeploymentsArray
-        {
-            private readonly BIP9DeploymentsParameters[] parameters;
-
-            public BIP9DeploymentsArray()
-            {
-                this.parameters = new BIP9DeploymentsParameters[Enum.GetValues(typeof(BIP9Deployments)).Length];
-            }
-
-            public BIP9DeploymentsParameters this[BIP9Deployments index]
-            {
-                get => this.parameters[(int)index];
-                set => this.parameters[(int)index] = value;
-            }
-        }
+        /// <inheritdoc />
+        public Money ProofOfStakeReward { get; }
+
+        /// <inheritdoc />
+        public uint MaxReorgLength { get; }
+
+        /// <inheritdoc />
+        public long MaxMoney { get; }
+
+        public ConsensusOptions Options { get; set; }
 
-        public Consensus()
+        public Consensus(IConsensus consensus)
         {
             this.BuriedDeployments = new BuriedDeploymentsArray();
             this.BIP9Deployments = new BIP9DeploymentsArray();
 
-            this.ConsensusFactory = new ConsensusFactory();
+            this.CoinbaseMaturity = consensus.CoinbaseMaturity;
+            this.PremineReward = consensus.PremineReward;
+            this.PremineHeight = consensus.PremineHeight;
+            this.ProofOfWorkReward = consensus.ProofOfWorkReward;
+            this.ProofOfStakeReward = consensus.ProofOfStakeReward;
+            this.MaxReorgLength = consensus.MaxReorgLength;
+            this.MaxMoney = consensus.MaxMoney;
+            this.Options = consensus.Options;
+            this.BuriedDeployments[NBitcoin.BuriedDeployments.BIP34] =
```

Comment:
Please configure VS to disable splitting the line when >80 chars. Lines up to 120-160 lines are ok.